<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Management and Exploratory Analysis of Spatial Data in Landscape Genetics — EcoGenetics-package • EcoGenetics</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">EcoGenetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/cran/EcoGenetics">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Management and Exploratory Analysis of Spatial Data in Landscape Genetics</h1>
    </div>

    
    <p>Management and exploratory analysis of spatial data in landscape genetics. Easy integration of information from multiple sources with "ecogen" objects.</p>
    

        
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p></p><table><tr><td>
 Package: </td>
<td> EcoGenetics</td>
</tr><tr><td>
 Type: </td>
<td> Package</td>
</tr><tr><td>
 Version: </td>
<td> 1.2.1-1</td>
</tr><tr><td>
 Date: </td>
<td> 2017-11-14</td>
</tr><tr><td>
 License: </td>
<td> GPL (&gt;=2) </td>
</tr></table>
    <p>----------------------- o-----------------------</p> 
<p><b><em>I. STRUCTURE OF THE PACKAGE</em></b></p>
<p>EcoGenetics has four basic modules. The <strong>base module</strong> is composed by general functions (<a href='eco.lmtree.html'>multiple lm</a>, <a href='eco.detrend.html'>detrending spatial data utility</a>, etc.). 
The <strong>general spatial module</strong> computes global (<a href='eco.gsa.html'>Moran's I</a>, <a href='eco.mantel.html'>Mantel test</a>, etc.) and local (<a href='eco.lsa.html'>Getis-Ord's G*</a>, <a href='eco.lsa.html'>local Moran's I</a>, etc.) spatial tests. These analyses use a <a href='eco.weight.html'>spatial weights matrix</a>, provided by the  <strong>spatial weights module</strong>.  The <strong>lag analysis module</strong> performs two basic analyses: the obtention  of <a href='eco.variogram.html'>variograms</a> and <a href='eco.correlog.html'>correlograms</a> (see also <a href='eco.cormantel.html'>this link</a>). This module uses the other tool provided by the spatial weights module: a sequence of <a href='eco.lagweight.html'>spatial weights matrices</a>.</p>
<p>The package have also special plot methods, as <a href='rankplot-methods.html'>rankplot</a>,  
<a href='forestplot-methods.html'>forestplot</a>, <a href='eco.rasterplot.html'>rasterplot</a> (implemented the last two for 
<a href='eco.lsa.html'>local spatial analysis</a>). Several conversor of data from/to other programs are available, (as to <a href='ecogen2genepop.html'>genepop</a> - an <a href='genepop2ecogen.html'>importer</a> tool is also defined for genepop-, <a href='ecogen2spagedi.html'>SPAGeDi</a>, etc.). Basic manipulation of genetic matrices is allowed by <a href='eco.convert.html'>eco.convert</a> and <a href='eco.format.html'>eco.format</a>. Tools for computation of NDVI in Landsat imaginery, 
post-process of rasters and temporal analysis can be found in <a href='eco.NDVI.html'>eco.NDVI</a>, <a href='eco.NDVI.post.html'>eco.NDVI.post</a> and <a href='eco.theilsen.html'>eco.theilsen</a>. Other useful functions are <a href='aue.sort.html'>aue.sort</a> (for ordering alleles), <a href='eco.alfreq.html'>eco.alfreq</a> (to comput histograms of allelic frequencies for detection of bottlenecks), <a href='eco.post.geneland.html'>eco.post.geneland</a> and <a href='eco.pairtest.html'>eco.pairtest</a>.</p>
<p>The results obtained with the main functions defined in EcoGenetics are
object of class S4. As a default characteristic of the package design,
these objects have a "show"" method for a general overview of the results, and methods to extract the results stored in slots (generic <a href='EcoGenetics-accessors.html'>accessors</a> and double square brackets ("[[") definitions).</p>
<p>For storing and pre-processing the data to analyze, the package 
defines a special class: the class <code><a href='ecogen.html'>ecogen</a></code>.</p> 
<p><b><em>II. STRUCTURE OF ECOGEN OBJECTS: HANDLING AND INTEGRATING INFORMATION</em></b></p>
    <p>Landscape genetics research requires the integration of data originated in different sources. The class <code><a href='ecogen.html'>ecogen</a></code> has been designed for handling multidimensional data. Its basic structure is the following:</p>
<p>- An <strong>XY</strong> slot, storing a data frame with geographic coordinates.</p>
<p>- A <strong>P</strong> slot, storing a phenotypic data frame.</p>
<p>- A <strong>G</strong> slot, storing a genotypic data frame.</p>
<p>- An <strong>A</strong> slot containing as allelic frequencies the information of G (only
available for codominant markers.)</p>
<p>- An <strong>E</strong> slot, storing an environmental data frame.</p>
<p>- A <strong>S</strong> slot, storing a data frame with classes assigned to the 
individuals.</p>
    <p>- A <strong>C</strong> slot, for a custom data frame.</p>
<p>- An <strong>OUT</strong> slot, containing a list for the storage of the results.</p>
    <p>For dominant (presence-absence) markers, the slot A is empty.</p>
<p><b><em>III. A BRIEF OVERVIEW OF THE DATA HANDLING METHODS DEFINED FOR ECOGEN OBJECTS</em></b></p>
    <p>The construction of a new "ecogen" object from a data frame is made with the homonymous function.
    <em>library("EcoGenetics")</em>
    <em>data(eco.test)</em>
    <em>eco &lt;- ecogen(XY = coordinates, P = phenotype, G = genotype, E = environment, S = structure, order.G = TRUE)</em></p>
    <p># The following methods can be used with ecogen objects:</p>
<p># - - - - - -
    <b># 1. subsetting/ordering by row-method, using single square brackets ("[")</b>
    <em>eco.sub &lt;- eco[1:50]</em>
    <em>eco.ord &lt;- eco[c(50:1)]</em>
    <em>eco.ord[["XY"]] # object ordered and subset rows.
This is a powerful method for handling the stack of ecogen data frames</em>
    <em>eco &lt;- eco[order(eco[["S"]][, 1])] #ordering S[, 1] and then the object</em>
    <b># 2. <a href='eco.merge.html'>merging-method</a>, for two objects</b>
    <em>eco1 &lt;- eco</em>
    <em>merged &lt;- eco.merge(eco, eco1)</em>
    <b># 3. <a href='eco.subset.html'>subsetting-method</a>, in reference to a group in the S slot (in this case, "1")</b>
    <em>eco.subS &lt;- eco.subset(eco,"pop", 1)</em></p>
<p><b># 4. <a href='eco.rbind.html'>binding by row-method</a> (duplicated row names not allowed)</b>
    <em>eco2 &lt;- eco; rownames(eco2[["P"]]) &lt;-226:450</em>
    <em>eco.r &lt;- eco.rbind(eco, eco2)</em>
    <b># 5. <a href='eco.cbind.html'>binding by column-method</a></b>
    <em>eco.c &lt;- eco.cbind(eco, eco1)</em></p>
<p><b># 6. get-method using generic <a href='EcoGenetics-accessors.html'>accessors</a> and "[[" (equivalent methods):</b>
    <em>ecoslot.XY(eco) ; eco[["XY"]]</em>
    <em>ecoslot.P(eco); eco[["P"]]</em>
    <em>ecoslot.G(eco); eco[["G"]]</em>
    <em>ecoslot.A(eco); eco[["A"]]  (only available for codominant data</em>
    <em>ecoslot.E(eco); eco[["E"]]</em>
    <em>ecoslot.S(eco); eco[["S"]]</em>
    <em>ecoslot.C(eco); eco[["C"]]</em>
    <em>ecoslot.OUT(eco); eco[["OUT"]]</em></p>
<p><b># 7. set-method using <a href='EcoGenetics-accessors.html'>accessors</a> and "[[" (equivalent methods):</b>
    <em>eco.temp &lt;- ecogen(XY = coordinates, P = phenotype)</em>
    <em>eco.temp</em>
    <em>ecoslot.G(eco.temp, order.G = TRUE) &lt;- genotype</em></p>
<p>## this is equivalent, in square brackets notation, to:
    <em>eco[["G", order.G=TRUE]] &lt;- genotype</em>
    <em>ecoslot.E(eco.temp) &lt;- environment</em></p>
<p>## identical to eco[["E"]] &lt;- environment
    <em>ecoslot.S(eco.temp) &lt;- structure</em></p>
<p>## identical to eco[["S"]] &lt;- structure
    <em>eco.temp</em>
    <b># 8. appending-method (storing information generated via <a href='EcoGenetics-accessors.html'>accessors)</a></b></p>
<p>## fitting a multiple linear regression model:
    <em>linear.analysis &lt;- eco.lmtree(eco[["P"]], eco[["E"]], "mlm")</em></p>
<p>## storing the results:
    <em>ecoslot.OUT(eco) &lt;- linear.analysis</em>
    <em>eco</em></p>
<p>## Storing multiple result at once:
    <em>a &lt;- c(1:10)</em>
    <em>b &lt;- c(2:30)</em>
    <em>ecoslot.OUT(eco) &lt;- list(eco, a, b)</em>
    <em>eco</em></p>
<p>## the use of the accessor OUT has its equivalence in double
square brackets notation:
    <em>eco[["OUT"]] &lt;- list(eco, a, b)</em></p>
    <p>## summary table
    <em>ecoslot.OUT(eco)</em></p>  
<p>## the data frame shows the results stored in alphabetical order and their classes. The specification of a second name, return the corresponding stored object:
    <em>ecoslot.OUT(eco, "a")</em></p>
<p>## note that the append method is a particular case of the
## set method with accessors / "[[" for the slot OUT.
    <b># 9. <a href='eco.remove.html'>removing-method</a></b></p>
<p>## removing objects a and b from eco
    <em>eco &lt;- eco.remove(eco, a, b)</em></p>
    <p>- - - - - -</p>
<p>The workspace can be <a href='eco.clear.html'>cleared</a>, only storing the desired object:
    <em>ls()</em>
    <em>eco.clear(eco)</em>
    <em>ls()</em></p>
<p>The cleaner and compact option, is to save the object as RDS,
which is a way to save R objects as a file:
    <em>saveRDS(eco, "myObject.rds")</em></p>
<p>The object can be loaded in the future with the function readRDS:
    <em>eco &lt;- readRDS("myObject.rds")</em></p>
<p>----------------------- o -----------------------</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
    <p>Anselin L. 1995. Local indicators of spatial association-LISA. Geographical analysis. 27: 93-115.</p>
<p>Borcard D., F. Gillet, and P. Legendre. 2011. Numerical ecology with R. Springer Science &amp; Business Media.</p>
<p>Chander G., B. Markham, and D. Helder. 2009. Summary of current radiometric calibration coefficients for Landsat MSS, TM, ETM+, and EO-1 ALI sensors. Remote sensing of environment, 113: 893-903.</p>
<p>Chavez P. 1989. Radiometric calibration of Landsat Thematic Mapper multispectral images. Photogrammetric Engineering and Remote Sensing, 55: 1285-1294.</p>
<p>Chavez P. 1996. Image-based atmospheric corrections-revisited and improved. Photogrammetric engineering and remote sensing, 62: 1025-1035.</p>
<p>Double M., R. Peakall, N. Beck, and Y. Cockburn. 2005. Dispersal, philopatry, and infidelity: dissecting local genetic structure in superb fairy-wrens (Malurs cyaneus). Evolution 59: 625-635.</p>
<p>Dray S., and A., Dufour. 2007. The ade4 package: implementing the duality diagram for ecologists. Journal of statistical software, 22: 1-20.</p> 
<p>Freedman D., and P. Diaconis. 1981. On the histogram as a density estimator: L 2 theory. Probability theory and related fields, 57: 453-476.</p>  
<p>Geary R. 1954. The contiguity ratio and statistical mapping. The incorporated statistician, 115-146.</p>
<p>Getis A., and J. Ord. 1992. The analysis of spatial association by use of distance statistics. Geographical analysis, 24: 189-206.</p>
<p>Goslee S. 2011. Analyzing remote sensing data in R: the landsat package. Journal of Statistical Software, 43: 1-25.</p>
<p>Goudet J. 2005. Hierfstat, a package for R to compute and test hierarchical F-statistics. Molecular Ecology Notes, 5: 184-186.</p> 
<p>Guillot G., F. Mortier and A. Estoup. 2005. GENELAND: a computer package for landscape genetics. Molecular Ecology Notes, 5: 712-715.</p> 
<p>Jombart T. 2008. adegenet: a R package for the multivariate analysis of genetic markers. Bioinformatics, 24: 1403-1405.</p>
<p>Kalisz S., J. Nason, F.M. Handazawa, and S. Tonsor. 2001. Spatial population genetic structure in Trillium grandiflorum:  the roles of dispersal, mating, history, and selection.  Evolution 55: 1560-1568.</p>
<p>Legendre P., and L. Legendre. 2012. Numerical ecology. Third English edition .Elsevier Science, Amsterdam, Netherlands.</p>
<p>Lichstein J., T. Simons, S. Shriner, and K. Franzreb. 2002. Spatial autocorrelation and autoregressive models in ecology. Ecological monographs, 72: 445-463.</p>
<p>Loiselle B., V. Sork, J. Nason, and C. Graham. 1995. Spatial genetic structure of a tropical understory shrub, Psychotria officinalis (Rubiaceae). American Journal of Botany 1420-1425.</p>
<p>Moran P. 1950. Notes on continuous stochastic phenomena. Biometrika, 17-23.</p>
<p>Oden, N., 1984. Assessing the significance of a spatial correlogram. 
Geographical Analysis, 16: 1-16.</p>
<p>Oden N., and R. Sokal. 1986. Directional autocorrelation: an extension of spatial correlograms to two dimensions. Systematic Zoology, 35: 608-617.</p>
<p>Ord J., and A. Getis. 1995. Local spatial autocorrelation statistics: distributional issues and an application. Geographical analysis, 27: 286-306.</p>
<p>Reich R., R. Czaplewski and W. Bechtold. 1994. Spatial cross-correlation of undisturbed, natural shortleaf pine stands in northern Georgia. Environmental and Ecological Statistics, 1: 201-217.</p>
<p>Sokal R., and N. Oden 1978. Spatial autocorrelation in biology: 1. Methodology. Biological journal of the Linnean Society, 10: 199-228.</p>
<p>Sokal R., and N. Oden. 1978. Spatial autocorrelation in biology. 2. Some biological implications and four applications of evolutionary and ecological interest. Biological Journal of the Linnean Society, 10: 229-49.</p>
<p>Sokal R. 1979. Ecological parameters inferred from spatial correlograms. In: G. Patil and M. Rosenzweig, editors. Contemporary Quantitative Ecology and elated Ecometrics. International Co-operative Publishing House: Fairland, MD, pp. 167-96.</p>
<p>Sokal R. 1986. Spatial data analysis and historical processes. In: E. Diday, Y. Escoufier, L. Lebart, J. Pages, Y. Schektman, and R. Tomassone, editors. Data analysis and informatics, IV. North-Holland, Amsterdam, The Netherlands, pp. 29-43.</p>
<p>Sokal R., N. Oden and B. Thomson. 1998. Local spatial autocorrelation in a biological model. Geographical Analysis, 30: 331-354.</p>
<p>Sokal R., and B. Thomson. 2006. Population structure inferred by local spatial autocorrelation: an example from an Amerindian tribal population. American journal of physical anthropology, 129: 121-131.</p>
<p>Song C., C. Woodcock, K. Seto, M. Lenney and S. Macomber. 2001. Classification and change detection using Landsat TM data: when and how to correct atmospheric effects?. Remote sensing of Environment, 75: 230-244.</p>
<p>Sturges H. 1926. The choice of a class interval. Journal of the American Statistical Association, 21: 65-66.</p>
<p>Tucker C. 1979. Red and photographic infrared linear combinations for monitoring vegetation. Remote sensing of Environment, 8: 127-150.</p>
<p>Vekemans, X., and O. Hardy. 2004. New insights from fine-scale spatial genetic structure analyses in plant populations. Molecular Ecology, 
13: 921-935.</p>
<p>Wu. 1986. Jackknife, bootstrap and other resampling methods in regression analysis. the Annals of Statistics, 1261-1295.</p>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      
      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>
          </ul>

    <h2>Author</h2>
    

 Leandro Roser, Juan Vilardi, Beatriz Saidman and Laura Ferreyra
 
 Maintainer: Leandro Roser &lt;leandroroser@ege.fcen.uba.ar&gt;

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Leandro Roser, Juan Vilardi, Beatriz Saidman, Laura Ferreyra.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
